berry/cli.py

```diff
@@ -59,58 +59,86 @@ def run_berry(args):
-                body = response['Body']
-                json_data = body.read()
-
-                # check that the file contains valid JSON
-                new_data = json.loads(json_data.decode('utf-8'))
-
-                try:
-                    with open(local_file, 'r') as fd:
-                        old_data = json.load(fd)
-                except:
-                    old_data = None
-                # check whether the file contents changed
-                if new_data != old_data:
-                    with open(tmp_file, 'wb') as fd:
-                        fd.write(json_data)
-                    os.rename(tmp_file, local_file)
-                    logging.info('Rotated {} credentials for {}'.format(fn, application_id))
-            except botocore.exceptions.ClientError as e:
-                # more friendly error messages
-                # https://github.com/zalando-stups/berry/issues/2
-                status_code = e.response.get('ResponseMetadata', {}).get('HTTPStatusCode')
-                msg = e.response['Error'].get('Message')
-                if status_code == 403:
-                    logging.error(('Access denied while trying to read "{}" from mint S3 bucket "{}". ' +
-                                   'Check your IAM role/user policy to allow read access! ' +
-                                   '(S3 error message: {})').format(
-                                  key_name, mint_bucket, msg))
-                elif status_code == 404:
-                    logging.error(('Credentials file "{}" not found in mint S3 bucket "{}". ' +
-                                   'Mint either did not sync them yet or the mint configuration is wrong. ' +
-                                   '(S3 error message: {})').format(
-                                  key_name, mint_bucket, msg))
-                else:
-                    logging.error('Could not read from mint S3 bucket "{}": {}'.format(
-                                  mint_bucket, e))
+                response = None
+                retry = 3
+                while retry:
+                    try:
+                        response = s3.get_object(Bucket=mint_bucket, Key=key_name)
+                        retry = False
+                    except botocore.exceptions.ClientError as e:
+                        # more friendly error messages
+                        # https://github.com/zalando-stups/berry/issues/2
+                        status_code = e.response.get('ResponseMetadata', {}).get('HTTPStatusCode')
+                        msg = e.response['Error'].get('Message')
+                        error_code = e.response['Error'].get('Code')
+                        endpoint = e.response['Error'].get('Endpoint', '')
+                        retry -= 1
+                        if error_code == 'InvalidRequest':
+                            # Try with a SigV4 Region
+                            region = 'eu-central-1'
```

Comment:
What about more regions with sigv4? Having the region hardcoded here sounds wrong